src/sonic-yang-models/yang-models/sonic-pbh.yang

```diff
@@ -0,0 +1,264 @@
+module sonic-pbh {
+
+	yang-version 1.1;
+
+	namespace "http://github.com/Azure/sonic-pbh";
+	prefix pbh;
+
+	import ietf-inet-types {
+		prefix inet;
+	}
+
+	import sonic-port {
+		prefix port;
+	}
+
+	import sonic-portchannel {
+		prefix lag;
+	}
+
+	description "PBH YANG Module for SONiC OS: hashing for NVGRE & VxLAN with IPv4/IPv6 inner 5-tuple";
+
+	revision 2021-04-23 {
+		description "First Revision";
+	}
+
+	typedef hash-field {
+		description "Represents native hash field";
+		type enumeration {
+			enum INNER_IP_PROTOCOL;
+			enum INNER_L4_DST_PORT;
+			enum INNER_L4_SRC_PORT;
+			enum INNER_DST_IPV4;
+			enum INNER_SRC_IPV4;
+			enum INNER_DST_IPV6;
+			enum INNER_SRC_IPV6;
+		}
+	}
+
+	typedef packet-action {
+		description "Represents packet action";
+		type enumeration {
+			enum SET_ECMP_HASH;
+			enum SET_LAG_HASH;
+		}
+	}
+
+	typedef flow-counter {
+		description "Represents flow counter state";
+		type enumeration {
+			enum DISABLED;
+			enum ENABLED;
+		}
+	}
+
+	container sonic-pbh {
+
+		container PBH_HASH_FIELD {
+
+			description "PBH_HASH_FIELD part of config_db.json";
+
+			list PBH_HASH_FIELD_LIST {
+
+				key "hash_field_name";
+
+				leaf hash_field_name {
+					description "The name of this hash field";
+					type string {
+						length 1..255;
+					}
+				}
+
+				leaf hash_field {
+					description "Configures native hash field for this hash field";
+					mandatory true;
+					type pbh:hash-field;
+				}
+
+				leaf ip_mask {
+					description "Configures IPv4/IPv6 address mask for this hash field";
+					mandatory true;
```

Comment:
Is mandatory "true" correct for this? eg: its not required for inner l4 src/dst ports and ip proto.

Reply:
@anish-n it's perfectly valid. Please refer to UTs. The trick here in using 'when' clause. Mandatory constraint takes place in case 'when' clause is satisfied.

Reply:
I see, thanks for explaining.

Reply:
@anish-n you're welcome.